pkg/source/reconciler/source/config_watcher.go

```diff
@@ -0,0 +1,160 @@
+package source
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/eventing-kafka/pkg/common/constants"
+	"knative.dev/eventing/pkg/reconciler/source"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/logging"
+)
+
+const (
+	EnvKafkaCfg           = "K_KAFKA_CONFIG"
+	kafkaConfigMapNameEnv = "CONFIG_KAFKA_NAME"
+)
+
+type KafkaConfig struct {
+	SaramaYamlString string
+}
+
+type KafkaSourceConfigAccessor interface {
+	source.ConfigAccessor
+	KafkaConfig() *KafkaConfig
+}
+
+// ToEnvVars serializes the contents of the ConfigWatcher to individual
+// environment variables.
+func (cw *KafkaSourceConfigWatcher) ToEnvVars() []corev1.EnvVar {
+	envs := cw.ConfigWatcher.ToEnvVars()
+
+	envs = maybeAppendEnvVar(envs, cw.kafkaConfigEnvVar(), cw.KafkaConfig() != nil)
+
+	return envs
+}
+
+// maybeAppendEnvVar appends an EnvVar only if the condition boolean is true.
+func maybeAppendEnvVar(envs []corev1.EnvVar, env corev1.EnvVar, cond bool) []corev1.EnvVar {
+	if !cond {
+		return envs
+	}
+
+	return append(envs, env)
+}
+
+// KafkaConfig returns the logging configuration from the ConfigWatcher.
+func (cw *KafkaSourceConfigWatcher) KafkaConfig() *KafkaConfig {
+	if cw == nil {
+		return nil
+	}
+	return cw.kafkaCfg
+}
+
+var _ KafkaSourceConfigAccessor = (*KafkaSourceConfigWatcher)(nil)
+
+type KafkaSourceConfigWatcher struct {
+	*source.ConfigWatcher
+	logger *zap.SugaredLogger
+
+	kafkaCfg *KafkaConfig
+}
+
+// WatchConfigurations returns a ConfigWatcher initialized with the given
+// options. If no option is passed, the ConfigWatcher observes ConfigMaps for
+// logging, metrics, tracing and Kafka.
+func WatchConfigurations(loggingCtx context.Context, component string,
+	cmw configmap.Watcher) *KafkaSourceConfigWatcher {
+
+	configWatcher := source.WatchConfigurations(loggingCtx, component, cmw)
+
+	cw := &KafkaSourceConfigWatcher{
+		ConfigWatcher: configWatcher,
+		logger:        logging.FromContext(loggingCtx),
+		kafkaCfg:      nil,
+	}
+
+	WithKafka(cw, cmw)
+
+	return cw
+}
+
+// WithKafka observes a Kafka ConfigMap.
+func WithKafka(cw *KafkaSourceConfigWatcher, cmw configmap.Watcher) {
```

Comment:
This function name is confusing to me; it gives me the impression that it is intended to be part of a builder paradigm (which would return another KafkaSourceConfigWatcher) but instead it's actively starting a watch.  Would suggest something like "WatchConfigMapWithKafka" or similar)